pkg/virt-controller/watch/vmi.go

```diff
@@ -515,6 +515,12 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 				}
 				vmiCopy.ObjectMeta.Labels[virtv1.NodeNameLabel] = pod.Spec.NodeName
 				vmiCopy.Status.NodeName = pod.Spec.NodeName
+
+				// Set the VMI migration transport now before the VMI can be migrated
+				// This status filed is needed to support the migration of legacy virt-launchers
+				// to newer ones. In an absence of this field on the vmi, the target launcher
+				// will set up a TCP proxy, as expected by a legacy virt-launcher.
+				vmiCopy.Status.MigrationTransport = virtv1.MigrationTransportUnix
```

Comment:
during a kubevirt update, couldn't setting the transport here result in a VMI running an outdated launcher pod being marked as capable of unix transport? 

For example,  take this sequence of events.

1. vmi is posted
2. kubevirt update is initiated
3. vmi pod is scheduling
4. new virt-controller marks MigrationTransport=unix even though old launcher pod was used.

This might not be that rare for situations where a VMI is waiting on resources to be scheduled.

Reply:
I think you are right. We should set it before the pod is created. Probably the best option would be to set it as an annotation on the pod when creating the pod and only reflecting it in the vm status then.